tools/mapping-tester/gatherstats.py

```diff
@@ -85,41 +101,58 @@ def memoryStats(dir):
     return stats
 
 
+def mappingStats(dir):
+    globber = os.path.join(dir, "*.stats.json")
+    statFiles = list(glob.iglob(globber))
+    if len(statFiles) == 0:
+        return {}
+
+    statFile = statFiles[0]
+    assert os.path.exists(statFile)
+    with open(os.path.join(dir, statFile), "r") as jsonfile:
+        return dict(json.load(jsonfile))
+
+
+def gatherCaseStats(casedir):
+    assert os.path.exists(casedir)
+    parts = os.path.normpath(casedir).split(os.sep)
+    assert len(parts) >= 5
+    mapping, constraint, meshes, ranks = parts[-4:]
+    meshA, meshB = meshes.split("-")
+    ranksA, ranksB = ranks.split("-")
```

Comment:
This part still feels pretty ridiculous and error-prone.

I'll save this information into a `case.json` in a future PR.